pennylane/ops/qubit.py

```diff
@@ -1910,6 +2008,9 @@ class BasisState(Operation):
     def decomposition(n, wires):
         return BasisStatePreparation(n, wires)
 
+    def adjoint(self, do_queue=False):
+        raise NotImplementedError
```

Comment:
I think it makes sense here to raise an `AdjointError: Adjoint is not supported for BasisState`. Partly because this doesn't feel like a not implemented error (which implies we can do it, it just hasn't been done).